lms/static/scripts/frontend_apps/api-types.ts

```diff
@@ -183,21 +183,21 @@ export type CoursesResponse = {
 export type Assignment = {
   id: number;
   title: string;
-  course: Course;
+  course: Course; // TODO Optional?
```

Comment:
This is indeed marked as optional in the backend types, so we should align the representation on the frontend:

```py
class APIAssignment(TypedDict):
    id: int
    title: str
    course: NotRequired[APICourse]
    annotation_metrics: NotRequired[AnnotationMetrics]
```

Reply:
I left this comment more to think if we should define two types.

Technically speaking, the prop could be defined as optional, but we know it's always there in some cases (when loading assignment metrics) and never there in other cases (when loading a list of assignments for the filters dropdown).

With two types we'll avoid extra conditional checks for the context in which we know the prop is there.